collect/queue.go

```diff
@@ -56,9 +56,21 @@ func Flush() {
 }
 
 func send() {
+
+	lastSent := time.Now()
+
 	for {
 		qlock.Lock()
 		if i := len(queue); i > 0 {
+
+			// If there is no backlog, wait for the minimum send interval before sending any more data
```

Comment:
I am confused with the logic here. If there is high volume of metrics you are likely to always have a backlog, thus continuing to send at high volume. Am I wrong?

Additionally, it looks like you throw away the last partial batch before sleeping. Should we not sleep only after sending?